src/components/RichLayout/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import Layout from 'components/Layout'
+import { LanguageFetcher } from 'components/Fetcher'
+import Header from './Header'
+import Navigation from 'Navigation'
+
+class HeaderAdapter extends React.Component {
+  static propTypes = {
+    location: PropTypes.string.isRequired
+  }
+
+  render () {
+    return <Header
+      languages={this.props.languages}
+      languageCallback={this.props.languageCallback}
+      currentLanguage={this.props.language}
+      navigation={new Navigation(this.props.location)}
+    />
+  }
+}
+
+class RichLayout extends React.Component {
+  static propTypes = {
+    className: PropTypes.string,
+    location: PropTypes.string.isRequired
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.gotoParent = this.gotoParent.bind(this)
+  }
+
+  gotoParent () {
+    // history.push(this.getParentPath()) fixme
+  }
+
+  render () {
+    return (<div>
+        <LanguageFetcher location={this.props.location} hideError={true} hideSpinner={true}>
+          <HeaderAdapter languageCallback={this.gotoParent} language={this.props.language} location={this.props.location}/>
+        </LanguageFetcher>
+
+        <Layout className={this.props.className}>
+          {this.props.children}
+        </Layout>
+      </div>
+    )
+  }
```

Comment:
Yes that makes sense but is an issue for another task. As soon as intelligent caching and refetching is implemented this won't be much of an issue.

I already noticed it when only the language button took some time to appear and this already was aweful. Therefore I think if it only happens once per location it actually looks better if the whole header takes some time to appear.

But I think this will be solved by the intelligent refreshing. But feel free to create another task after this merge.